src/MainWindow.cpp

```diff
@@ -0,0 +1,936 @@
+    m_amountAvailableLabel->setText(m_TOTAL_AMOUNT_AVAILABLE_STRING);
+
+    // Create the amount available line edit and set its location
+    m_amountAvailableEdit = new QLineEdit(this);
+    m_amountAvailableEdit->setGeometry(155, 0, 75, 20);
+
+    // Create the bill table widget and set its location
+    m_billTableWidget = new QTableWidget(this);
+    m_billTableWidget->setGeometry(0, 100, 500, 500);
+
+    // Create the save button
+    m_saveButton = new QPushButton(this);
+    m_saveButton->setText(m_SAVE_BUTTON_TEXT);
+
+    // When the Save button is clicked, update the bill map and config file to match the contents of the bill table widget
+    connect(m_saveButton, SIGNAL(clicked()), this, SLOT(updateConfigFromUI()), Qt::AutoConnection);
+
+    // Create the Add Another Bill button
+    m_addBillButton = new QPushButton(this);
+    m_addBillButton->setText(m_ADD_BILL_BUTTON_TEXT);
+
+    // When the Add Another Bill button is clicked, clear the bill widget and display it for entry of another bill
+    connect(m_addBillButton, SIGNAL(clicked()), this, SLOT(showBillWidget()), Qt::AutoConnection);
+
+    // Create the Delete button
+    m_deleteBillButton = new QPushButton(this);
+    m_deleteBillButton->setText(m_DELETE_BUTTON_TEXT);
+
+    // When the Delete button is clicked, make sure to update the amount available and funded bill lists if the deleted bill was funded, and then remove from the bill table widget
+    connect(m_deleteBillButton, SIGNAL(clicked()), this, SLOT(deleteBillOnClick()), Qt::AutoConnection);
+
+    // Create the Fund Bill button
+    m_fundBillButton = new QPushButton(this);
+    m_fundBillButton->setText(m_FUND_BILL_BUTTON_TEXT);
+
+    // When the Fund Bill button is clicked, switch the selected bill's funding status to Funded
+    connect(m_fundBillButton, SIGNAL(clicked()), this, SLOT(fundBillOnClick()), Qt::AutoConnection);
+
+    // Create the Defund Bill button
+    m_defundBillButton = new QPushButton(this);
+    m_defundBillButton->setText(m_DEFUND_BILL_BUTTON_TEXT);
+
+    // When the Defund Bill button is clicked, switch the selected bill's funding status to Not Funded
+    connect(m_defundBillButton, SIGNAL(clicked()), this, SLOT(defundBillOnClick()), Qt::AutoConnection);
+
+    // Organize the buttons into a grid layout
+    createButtonGridLayout();
+
+    // Kick off the start sequence where we check for the existence of the config file
+    attemptConfigFileGeneration();
+}
+
+MainWindow::~MainWindow()
+{
+    // If the bill widget has been created successfully
+    if(m_billWidget != nullptr)
+    {
+        // Delete it and set it to be null
+        delete m_billWidget;
+        m_billWidget = nullptr;
+    }
+
+    // Properly delete the amount available label
+    deleteLabelIfNonNull(m_amountAvailableLabel);
+
+    // Properly delete the amount available line edit
+    deleteLineEditIfNonNull(m_amountAvailableEdit);
+
+    // Properly delete the save button
```

Comment:
Change this comment to mention deleting all the buttons, not just the save button.

Reply:
Comment changed.